src/Widgets/TerminalWidget.vala

```diff
@@ -148,27 +149,23 @@ namespace Terminal {
                     get_clicked_cell_position (event, out clicked_row, out clicked_col);
                     remember_command_start_position ();
 
-                    if (clicked_row < remembered_command_start_row) {
+                    if (
+                        clicked_row < remembered_command_start_row
+                        || (clicked_row == remembered_command_start_row
+                        && clicked_col < remembered_command_start_col)
+                    ) {
                         return Gdk.EVENT_PROPAGATE;
                     }
 
                     long delta_cells = clicked_col - current_col + (clicked_row - current_row) * get_column_count ();
 
-                    /* Synthesise a cursor press - is there a better way? */
-                    Gdk.EventKey key_event = (Gdk.EventKey)(new Gdk.Event (Gdk.EventType.KEY_PRESS));
-                    key_event.send_event = 1;
-                    /* Need to add a ref else crash on second key press - vapi error? */
-                    key_event.window = (Gdk.Window)(this.get_window ().ref ());
-                    key_event.keyval = delta_cells > 0U ? Gdk.Key.Right : Gdk.Key.Left;
-                    key_event.is_modifier = 0;
-
                     set_color_cursor (background_color);
                     Idle.add (() => { /* wait for button press event to be processed */
                         /* Cursor will move as close as possible to pointer */
                         var n_events = (int) delta_cells.abs ();
+                        var sequence = delta_cells > 0U ? "\033[C" : "\033[D";
                         for (int i = 0; i < n_events; i++) {
-                            key_event.time = (uint32)(get_monotonic_time ());
-                            key_press_event (key_event);
+                            feed_child (sequence, 3);
                         }
```

Comment:
I think you can use a single sequence to move the cursor <N> places.  Either `\033[<N>C` or `\033[<N>D`

Reply:
I also thought so, but I did not have the desired effect. Instead it moves only one position and deletes the last character. Maybe I am doing something wrong... 

There is also a sequence to move the cursor to a fixed position `Esc[Line;ColumnH`. But I was also not able to make it work ...

Reply:
@jeremypw  Do you understand why on https://valadoc.org/vte-2.91/Vte.Terminal.feed_child.html the `feed_child` has a diffenerent signature (in comparison to `feed_child (string, int)`)